File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/RefreshImageListUseCaseImpl.java
package com.hwonchul.movie.domain.usecase.details;

import androidx.annotation.NonNull;

import com.hwonchul.movie.domain.repository.ImageRepository;

import javax.inject.Inject;

import io.reactivex.rxjava3.core.Completable;

public class RefreshImageListUseCaseImpl implements RefreshImageListUseCase {
    private final ImageRepository repository;

    @Inject
    public RefreshImageListUseCaseImpl(@NonNull ImageRepository repository) {
        this.repository = repository;
    }

    @Override
    public Completable invoke(int movieId) {
        return repository.refreshFromRemote(movieId);
    }
}

File: app/src/main/java/com/hwonchul/movie/domain/usecase/details/RefreshImageListUseCaseImpl.kt
package com.hwonchul.movie.domain.usecase.details

import com.hwonchul.movie.domain.repository.ImageRepository
import javax.inject.Inject

class RefreshImageListUseCaseImpl @Inject constructor(
    private val repository: ImageRepository
) : RefreshImageListUseCase {

    override suspend fun invoke(movieId: Int): Result<Unit> =
        runCatching { repository.refreshFromRemote(movieId) }
}